lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2568,6 +2568,98 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSliceTensorOp : public OpConversionPattern<AtenSliceTensorOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSliceTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = adaptor.self();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+    SmallVector<Value> resultShape;
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    SmallVector<Value> offsets, strides;
+    strides.resize(inputType.getRank(), one);
+    offsets.resize(inputType.getRank(), zero);
+    Value end, start, stepIndex;
+    resultShape = getTensorSizes(rewriter, loc, input);
+    int64_t dim, step;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return op->emitError("unimplemented: dim is not constant");
+    if (op.start().getType().isa<Torch::NoneType>()) {
+      start = zero;
+    } else {
+      auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+      Value startPositive =
+          toPositiveDimDynamic(rewriter, loc, adaptor.start(), temp);
+      // start < 0 ? 0 : start
+      Value cst0 = rewriter.create<arith::ConstantOp>(
+          loc, rewriter.getZeroAttr(temp.getType()));
+      Value predDimSltZero = rewriter.create<arith::CmpIOp>(
+          loc, arith::CmpIPredicate::slt, startPositive, cst0);
+      Value startAtLeastZero =
+          rewriter.create<SelectOp>(loc, predDimSltZero, cst0, startPositive);
+      start = castIntToIndex(rewriter, loc, startAtLeastZero);
+    }
+    if (op.end().getType().isa<Torch::NoneType>()) {
+      end = resultShape[dim];
+    } else {
+      auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+      end = toPositiveDimDynamic(rewriter, loc, adaptor.end(), temp);
+      end = castIntToIndex(rewriter, loc, end);
+    }
+    if (!matchPattern(op.step(), m_TorchConstantInt(&step))) {
+      if (!op.step().getType().isa<Torch::NoneType>())
+        return op->emitError("unimplemented: step is not constant");
+      step = 1;
+    }
+    // Slice logic: resultSize = floordiv(end - start + step - 1,  step)
```

Comment:
I think this won't work for cases where `end < start` can you add an runtime assertion `end > start`?

Reply:
we need to also assert `start < dimSize` and `end <= dimSize` with a message saying those cases are unimplemented.  And add TODOs.

Reply:
I added some code to do what pytorch does, lmk what you think